models/brats_mri_axial_slices_generative_diffusion/docs/README.md

```diff
@@ -0,0 +1,145 @@
+
+**This is a demonstration network meant to just show the training process for this sort of network with MONAI. To achieve better performance, users need to use larger dataset like [Brats 2021](https://www.synapse.org/#!Synapse:syn25829067/wiki/610865).**
+
+## Install the dependency of MONAI generative models
+[MONAI generative models](https://github.com/Project-MONAI/GenerativeModels) can be installed by
+```
+pip install lpips
+pip install git+https://github.com/Project-MONAI/GenerativeModels.git@0.2.0
+```
+
+Or we can install it from source
+```
+pip install lpips
+git clone https://github.com/Project-MONAI/GenerativeModels.git
+cd GenerativeModels/
+python setup.py install
+cd ..
+```
+
+## Downloading the Dataset
+The training data is Brats 2016 and 2017 from [Medical Decathlon](http://medicaldecathlon.com/).
+
+- Target: image generatiion
+- Task: Synthesis
+- Modality: MRI
+- Size: 388 3D MRI volumes (1 channel used)
+- Training data size: 38800 2D MRI axial slices (1 channel used)
+
+The dataset can be downloaded automatically at the beggining of training.
+
+## Training configuration
+If user has GPU memory smaller than 32G, then please decrease the `"train_batch_size_img"` and `"train_batch_size_slice"` in `configs/train_autoencoder.json` and `configs/train_diffusion.json`.
+`"train_batch_size_img"` is number of 3D volumes loaded in each batch. `"train_batch_size_slice"` is the number of 2D axial slices extracted from each image. The actual batch size is the product of them.
+
+### Training configuration of autoencoder
+The training of autoencoder was performed with the following:
+
+- GPU: at least 32GB GPU memory
+- Actual Model Input: 240 x 240
+- AMP: False
+- Optimizer: Adam
+- Learning Rate: 2e-5
+- Loss: L1 loss, perceptual loss, KL divergence loss, adversianl loss, GAN BCE loss
+
+#### Input
+1 channel 2D MRI Flair axial patches
+
+#### Output
+- 1 channel 2D MRI reconstructed patches
+- 1 channel mean of latent features
+- 1 channel standard deviation of latent features
+
+### Training configuration of diffusion model
+The training of latent diffusion model was performed with the following:
+
+- GPU: at least 32GB GPU memory
+- Actual Model Input: 64 x 64
+- AMP: False
+- Optimizer: Adam
+- Learning Rate: 1e-5
+- Loss: MSE loss
+
+#### Training Input
+- 1 channel noisy latent features
+- an int that indicates the time step
+
+#### Training Output
+1 channel predicted added noise
+
+#### Inference Input
+1 channel noise
+
+#### Inference Output
+1 channel denoised latent features
+
+
+## MONAI Bundle Commands
```

Comment:
## MONAI Bundle Commands
In addition to the Pythonic APIs, a few command line interfaces (CLI) are provided to interact with the bundle. The CLI supports flexible use cases, such as overriding configs at runtime and predefining arguments in a file.

For more details usage instructions, visit the [MONAI Bundle Configuration Page](https://docs.monai.io/en/latest/config_syntax.html).

#### Execute Autoencoder Training (w/data download)

Assuming the current directory is the bundle directory, the following command will train the autoencoder network for 1500 epochs using the BraTS dataset. If the dataset is not downloaded, it will be automatically downloaded and extracted to `./Task01_BrainTumour`.

```
python -m monai.bundle run --config_file configs/train_autoencoder.json --dataset_dir ./ --download_brats True
```

#### Execute Autoencoder Training
If the dataset is already downloaded, make sure that `"dataset_dir"` in `configs/train_autoencoder.json` has the correct path to the dataset `Task01_BrainTumour`. Then, run:

```
python -m monai.bundle run --config_file configs/train_autoencoder.json
```

#### Override the `train` config to execute multi-GPU training for Autoencoder:
To train with multiple GPUs, use the following command, which requires scaling up the learning rate according to the number of GPUs. Keep in mind that this command will take approximately 9 hours to complete when using 8 GPUs, each with 32G of memory.

```
torchrun --standalone --nnodes=1 --nproc_per_node=8 -m monai.bundle run --config_file "['configs/train_autoencoder.json','configs/multi_gpu_train_autoencoder.json']" --lr 2e-4
```

#### Execute Latent Diffusion Model Training
After training the autoencoder, run the following command to train the latent diffusion model. This command will print out the scale factor of the latent feature space. If your autoencoder is well trained, this value should be close to 1.0.

```
python -m monai.bundle run --config_file "['configs/train_autoencoder.json','configs/train_diffusion.json']"
```

#### Override the `train` config to execute multi-GPU training for Latent Diffusion Model:
To train with multiple GPUs, use the following command, which requires scaling up the learning rate according to the number of GPUs. Keep in mind that this command will take approximately 5 hours to complete when using 8 GPUs, each with 32G of memory.
```
torchrun --standalone --nnodes=1 --nproc_per_node=8 -m monai.bundle run --config_file "['configs/train_autoencoder.json','configs/train_diffusion.json','configs/multi_gpu_train_autoencoder.json','configs/multi_gpu_train_diffusion.json']"  --lr 8e-5
```


#### Execute inference:
The following code generates a synthetic image from a random sampled noise.
```
python -m monai.bundle run --config_file configs/inference.json
```
The generated image will be saved to `./output`

![Example synthetic image](https://developer.download.nvidia.com/assets/Clara/Images/monai_brain_image_gen_ldm2d_example_generation.png)